### PR TITLE
Describe programmatic Broccoli Builder usage in README

### PR DESCRIPTION
**Pull request by cafreeman**
This PR attempts to address https://github.com/broccolijs/broccoli/issues/368.

Let me know if I've gotten anything wrong, or if there are issues with the style and I'm happy to address.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,68 @@ running `broccoli build the-output` would generate the following folder:
...
+
+const outputDir = 'dist';
+const outputTree = new TreeSync(builder.outputPath, outputDir);
+
+builder.build()
+  .then(() => {
+    // Calling `sync` will synchronize the contents of the builder's `outPath` with our output directory.
+    outputTree.sync();
+  })
+  .finally(() => {
```
Comment (krisselden) [inline README.md]: finally does not exist until node 10

Comment (rwjblue) [reply README.md]: yes, but you would need to do it in both success and failure cases (e.g. `.then` + `.catch` or `.then(cb, cb)`)

Comment (cafreeman) [reply README.md]: Gotcha. Will do!

Comment (cafreeman) [reply README.md]: @rwjblue updated!

**Pull request merged**